pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -0,0 +1,1150 @@
+			case api.ResourceMemory:
+				podMemoryLimit = v
+			}
+		}
+		for k, v := range podResourceRequests {
+			switch k {
+			case api.ResourceCPU:
+				podCPURequest = v
+			case api.ResourceMemory:
+				podMemoryRequest = v
+			}
+		}
+		if podCPULimit != 0 || podCPURequest != 0 {
+			cpuResource := &runtimeApi.ResourceRequirements{}
+			if podCPULimit != 0 {
+				cpulimit := float64(podCPULimit) / 1000
+				cpuResource.Limits = &cpulimit
+			}
+			if podCPURequest != 0 {
+				cpuRequest := float64(podCPURequest) / 1000
+				cpuResource.Requests = &cpuRequest
+			}
+			podSandboxConfig.Resources.Cpu = cpuResource
+		}
+		if podMemoryLimit != 0 || podMemoryRequest != 0 {
+			memoryResource := &runtimeApi.ResourceRequirements{}
+			if podMemoryLimit != 0 {
+				memLimit := float64(podMemoryLimit) / 1000
+				memoryResource.Limits = &memLimit
+			}
+			if podMemoryRequest != 0 {
+				memRequest := float64(podMemoryRequest) / 1000
+				memoryResource.Requests = &memRequest
+			}
+			podSandboxConfig.Resources.Memory = memoryResource
+		}
+	}
+
+	return podSandboxConfig, nil
+}
+
+func (m *kubeGenericRuntimeManager) createPodSandbox(pod *api.Pod) (string, error, string) {
+	podSandboxConfig, err := m.generatePodSandboxConfig(pod)
+	if err != nil {
+		message := fmt.Sprintf("GeneratePodSandboxConfig failed: %v", err)
+		glog.Error(message)
+		return "", err, message
+	}
+
+	podSandBoxID, err := m.runtimeService.CreatePodSandbox(podSandboxConfig)
+	if err != nil {
+		message := fmt.Sprintf("CreatePodSandbox failed: %v", err)
+		glog.Error(message)
+		return "", err, message
+	}
+
+	return podSandBoxID, nil, ""
+}
+
+func (m *kubeGenericRuntimeManager) makeMounts(containerFullName string, opts *kubecontainer.RunContainerOptions, container *api.Container) []*runtimeApi.Mount {
+	volumeMounts := []*runtimeApi.Mount{}
+
+	for _, v := range opts.Mounts {
+		volumeMounts = append(volumeMounts, &runtimeApi.Mount{
+			Name:           &v.Name,
+			HostPath:       &v.HostPath,
+			ContainerPath:  &v.ContainerPath,
+			Readonly:       &v.ReadOnly,
+			SelinuxRelabel: &v.SELinuxRelabel,
+		})
+	}
+
+	if opts.PodContainerDir != "" && len(container.TerminationMessagePath) != 0 {
+		containerLogPath := path.Join(opts.PodContainerDir, containerFullName)
+		fs, err := os.Create(containerLogPath)
+		if err != nil {
+			glog.Errorf("Error on creating termination-log file %q: %v", containerLogPath, err)
+		} else {
+			fs.Close()
+			volumeMounts = append(volumeMounts, &runtimeApi.Mount{
+				HostPath:      &containerLogPath,
+				ContainerPath: &container.TerminationMessagePath,
+			})
+		}
+	}
+
+	return volumeMounts
+}
+
+func (m *kubeGenericRuntimeManager) makeContainerLogsPath(containerName, podUID string) string {
+	return path.Join(containerLogsDir, podUID, fmt.Sprintf("%s.log", containerName))
+}
+
+func (m *kubeGenericRuntimeManager) generateContainerConfig(container *api.Container, pod *api.Pod, restartCount int) (*runtimeApi.ContainerConfig, error) {
```

Comment:
can we also make this a helper method (not part of kubeGenericRuntimeManager) and move it to a separate file with the `makeMounts`? This would make this file more readable IMO.